File: parent-group4/pingyougou-interface/src/main/java/com/itheima/core/SpecificationService.java
package com.itheima.core;

import com.itheima.core.pojo.specification.Specification;
import entity.PageResult;
import pojogroup.SpecificationVo;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    PageResult search(Integer page, Integer rows, Specification specification);

    void add(SpecificationVo specificationVo);

    SpecificationVo findOne(Long id);

    void update(SpecificationVo specificationVo);

    void delete(Long[] ids);

    List<Map> selectOptionList();
}

File: parent-group4/pingyougou-web-shop/src/main/webapp/js/service/specificationService.js
//服务层
app.service('specificationService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../specCheck/findAll.do');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../specCheck/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../specCheck/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../specCheck/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../specCheck/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../specCheck/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../specCheck/search.do?page='+page+"&rows="+rows, searchEntity);
	}  
	
	this.selectOptionList=function(){
		return $http.get("../specCheck/selectOptionList.do");
	}
});
